pkg/indexer/localcredentials.go

```diff
@@ -0,0 +1,83 @@
+package indexer
+
+import (
+	"context"
+	"reflect"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+)
+
+const (
+	AtlasDatabaseUserCredentialsIndex = "atlasdatabaseuser.credentials"
+)
+
+type LocalCredential struct {
```

Comment:
Given the comment from @s-urbaniak about the project ref indexer, I'm wondering a generic indexer from local secret is a good practice.

Reply:
I really do not want to pi¡y&paste and rename an indexer for each local cred usage.

Reply:
I understand your motivation, but most of our indexes follow a pattern to refer to a project or a secret, we could group and easily generalize the implementation for each case, however, we have an indexer per index. Why should we deviate from this here?

Reply:
Less code to write, to test and less potential mistakes.

The current fix covers the initial concerns and will support all other usages of the credentials by just adding a single mapping function, similar to `DatabaseUserRequests`.